notebooks/passenger_vehicle_co2_lca.py

```diff
@@ -0,0 +1,322 @@
+
+# ## Definiera bilar
+# För att kunna göra jämförelsen behöver vi välja vilka biltyper vi vill titta på. Följande tre typexempel har tagits fram:
+#
+# ##### Version 1
+# |Bil | Vikt exkl batteri (kg) | Batteri (kWh) | Räckvidd el (km) | Energiförbrukning| Användande|
+# |---|---|---|---|---|---|
+# |A: Liten bensinbil | 1500 | - | - | 0.5 liter bensin/mil |100%|
+# |B: Stor elbil | 2200 | 125 | 500 | 0.25 kWh/km |100%|
+# |C: Liten elbil | 1500 | 50 | 330 | 0.15 kWh/km |100%|
+# |D: Cykel | 10 | 0 | 0 | 0kWh/km |100%|
+# |E: Elcykel | 15 |0.5 | 20 | 0.005 kWh/km |100%|
+#
+
+cars = {
+    "A: Liten bensinbil": [1500, 0, 0, 0.05, 1],
+    "B: Stor elbil": [2200, 125, 500, 0.25, 1],
+    "C: Liten elbil": [1500, 50, 330, 0.15, 1],
+    "D: Cykel": [10, 0, 0, 0, 1],
+    "E: Elcykel": [15, 0.5, 20, 0.005, 1],
+}
+print(cars)
+
+
+# Definiera funktion som ger CO2 över tid
+def co2analysis(cars, driven_distance, co2debt):
+    """CO2 accumulator function."""
+    car_data = []
+    for key, value in cars.items():
+        C02_data_constant = (
+            co2debt
+            + car_manufacturing_cost * value[0]
+            + battery_manufacturing_cost * value[1]
+        )
+        if value[1] == 0:
+            C02_data = (
+                C02_data_constant
+                + kgC02_per_litre
+                * value[3]
+                * (driven_distance - driven_distance[0])
+                * value[4]
+            )
+        else:
+            C02_data = (
+                C02_data_constant
+                + kgC02_per_kWh_europe
+                * value[3]
+                * (driven_distance - driven_distance[0])
+                * value[4]
+            )
+        car_data.append([key, driven_distance, C02_data / 1000])
+
+    return car_data
+
+
+driven_distance = np.linspace(0, 300000, 200)
+leg = []
+car_data = co2analysis(cars, driven_distance, 0)
+for value in car_data:
+    plt.plot(value[1], value[2])
+    leg.append(value[0])
+plt.legend(leg)
+plt.xlabel("kördistans [km]")
+plt.ylabel("CO2 [metric tonnes]")
+plt.show()
+
+# ## Fordonspark
+# Många av oss äger ju flera fordon. Hur mycket förbättrar vi situationen om vi cyklar 10km för varje 100km vi kör i vår stora bil? Här jämför vi tre alternativ för omläggning av persontransporterna för någon som tidigare kört en liten bensinbil men som nu vill lägga om sitt liv för att minska klimatpåverkan. Efter 16667mil bilpendling bestämmer sig vår protagonist för att lägga om sitt liv för att minska sitt klimatavtryck, genom att:
+#
+# 1. Fortsätta köra samma bensinbil
+# 2. Köpa en stor elbil
+# 3. Köpa en liten elbil
+# 4. Köpa en stor elbil och en cykel
+# 5. Köpa en stor elbil och en elcykel
+# 6. Köpa en elcykel
+#
+# Låt oss definiera en brytpunkt till att börja med:
+
+kilometers_at_change = (
+    166667  # Average current driven distance according to analysis below
+)
+
+# +
+# Scenario 1: Fortsätta köra samma bensinbil
```

Comment:
I would like to see a loop here, instead of a wall of code. Try to create cars such that you can loop over it.

Reply:
Yes, I will add a back-end.